src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -569,6 +567,80 @@ private void OutputResultsHelper(IEnumerable<CommandInfo> results)
 #endif
         }
 
+        /// <summary>
+        /// Creates the syntax output based on if the command is an alias, script, application or command.
+        /// </summary>
+        /// <param name="command">
+        /// CommandInfo object containing the syntax to be output.
+        /// </param>
+        /// <returns>
+        /// Syntax string cast as a PSObject for outputting.
+        /// </returns>
+        private PSObject GetSyntaxObject(CommandInfo command)
+        {
+            PSObject syntax = PSObject.AsPSObject(command.Syntax);
+
+            if (this.Name != null && !Array.Exists(this.Name, name => name.Equals(command.Name, StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
Perhaps it could be easier to just do a `foreach` over the `Name` values and work with each separately?

Reply:
There's already a foreach loop in the `OutputResultsHelper` method which sends one found command at a time to the output, via this new method if the `Syntax` switch is provided. So I'm working with one at a time but need to check if the command I'm being passed was declared as the actual command to find or an alias or a wildcard, so I can then output things in the correct way.

Reply:
So what's this condition meant to be checking? Just not sure why we're using Array.Exists here & wondering if the condition can be simplified. 🙂

Reply:
It's checking if the command name that's been passed in is one that was specified by a user, if not then we have to assume they specified an alias or a wildcard and do some extra formatting for those, if it is then just go with the default formatting. 

So if a user runs `Get-Command -Name del -Syntax` the code will find del and the command it resolves to as Remove-Item and attempt to return that, but as the user specified del we want to fiddle with the output a bit to make it clear that's an alias but still give the Remove-Item syntax.

Reply:
Could you use `Array.IndexOf(Name, command.Name) > -1`? I don't know for sure, but I have a feeling using the predicate might be a tad slower, and it isn't very clear in terms of the intent of the condition imo. 🙂